connect-inject/container_env_test.go

```diff
@@ -0,0 +1,78 @@
+package connectinject
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestContainerEnvVars(t *testing.T) {
+	serviceHost := "static-server"
+	servicePort := "7890"
+	upstreamAnnotation := fmt.Sprintf("%s:%s:dc2", serviceHost, servicePort)
+
+	minimal := func() *corev1.Pod {
+		return &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotationService: "foo",
+				},
+			},
+		}
+	}
+
+	cases := []struct {
+		Name        string
+		Pod         func(*corev1.Pod) *corev1.Pod
+		ServiceHost string // Service host name used to craft env var names
+		ServicePort string // Service port that should be set to env var
+	}{
```

Comment:
I'd recommend just hard-coding the upstream values and also adding a testcase for an upstream without datacenter. I suggest changing this to:
```go
cases := []struct {
  name string
  upstream string
}{
  "Upstream with datacenter",
  "static-server:7890:dc1",
},
{
  "Upstream without datacenter",
  "static-server:7890",
}
```
Then you could move the minimal pod inside the `for` loop and remove the `serviceHost`, `servicePort`, and `upstreamAnnotation` variables.

```go

	for _, tt := range cases {
		t.Run(tt.Name, func(t *testing.T) {
			require := require.New(t)

			var h Handler
			envVars := h.containerEnvVars(&corev1.Pod{
			ObjectMeta: metav1.ObjectMeta{
				Annotations: map[string]string{
					annotationService: "foo",
                                        annotationUpsteams: tt.upstream,
				},
			      },
		           }
	                })

			....
}
```

Reply:
Thanks for the feedback @ishustava .  I'll take a stab in my off-hours and ping when I've updated.

Reply:
@ishustava  I made the modifications.  Thanks for the input.  Simplified things a lot.